src/ResourceManager/KeyVault/KeyVaultProvider/KeyVaultProvider.cs

```diff
@@ -0,0 +1,3365 @@
+                        var filteredPaths = new List<string>();
+                        var allStrings = kvDriveInfo.KeyVaultDataServiceClient.GetCertificates(
+                            new KeyVaultCertificateFilterOptions { VaultName = namesFromPath[0], NextLink = null });
+
+                        foreach (var stringItem in allStrings)
+                        {
+                            if (stringItem.Name.StartsWith(filter))
+                            {
+                                int separatorIndex = path.LastIndexOf(pathSeparator);
+                                var trimmedPath = path.Substring(0, separatorIndex);
+                                filteredPaths.Add(trimmedPath + pathSeparator + stringItem.Name);
+                            }
+                        }
+                        return filteredPaths.ToArray();
+                    }
+                    else if (namesFromPath[1] == "Keys")
+                    {
+                        var filter = namesFromPath[2].TrimEnd('*');
+                        var filteredPaths = new List<string>();
+                        var allStrings = kvDriveInfo.KeyVaultDataServiceClient.GetKeys(
+                            new KeyVaultObjectFilterOptions { VaultName = namesFromPath[0], NextLink = null });
+
+                        foreach (var stringItem in allStrings)
+                        {
+                            if (stringItem.Name.StartsWith(filter))
+                            {
+                                int separatorIndex = path.LastIndexOf(pathSeparator);
+                                var trimmedPath = path.Substring(0, separatorIndex);
+                                filteredPaths.Add(trimmedPath + pathSeparator + stringItem.Name);
+                            }
+                        }
+                        return filteredPaths.ToArray();
+                    }
+                    else if (namesFromPath[1] == "AccessPolicies")
+                    {
+                        var resourceGroupName = this.GetResourceGroupName(namesFromPath[0]);
+                        PSKeyVault vault = null;
+                        if (resourceGroupName != null)
+                        {
+                            vault = kvDriveInfo.KeyVaultClient.GetVault(namesFromPath[0], resourceGroupName, kvDriveInfo.ActiveDirectoryClient);
+                        }
+
+                        var filter = namesFromPath[2].TrimEnd('*');
+                        var filteredPaths = new List<string>();
+                        var allStrings = vault.AccessPolicies;
+
+                        foreach (var stringItem in allStrings)
+                        {
+                            if (stringItem.ObjectId.StartsWith(filter))
+                            {
+                                int separatorIndex = path.LastIndexOf(pathSeparator);
+                                var trimmedPath = path.Substring(0, separatorIndex)
```

Comment:
@maddieclayton nit: this can just be `if (vault.EnableSoftDelete)`

It looks like you do this in a few places throughout this class, so just make sure those are addressed as well